src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -132,7 +132,7 @@ public void onResolveFailed(NsdServiceInfo serviceInfo, int errorCode) {
                 "Failed to resolve service '" + serviceInfo.getServiceName() + "': " + errorCode);
             chipMdnsCallback.handleServiceResolve(
                 serviceInfo.getServiceName(),
-                serviceInfo.getServiceType(),
+                NsdServiceFinderAndResolver.this.targetServiceInfo.getServiceType(),
```

Comment:
this seems like it should be a separate PR with separate description

Reply:
I'm sorry. I wasn't crazy enough to check it out. This code was previously requested as #25645. It has not been merged yet, so it seems to have been added together by mistake.